File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/adapter/SimpleRecyclerViewAdapter.java
package tv.teads.teadssdkdemo.format.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import java.util.List;

import tv.teads.sdk.android.InReadAdView;
import tv.teads.teadssdkdemo.R;

/**
 * Simple RecyclerView adapter
 * <p/>
 * Created by Hugo Gresse on 08/07/15.
 */
public class SimpleRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final int TYPE_TEADS = 0;
    private static final int TYPE_TEXT  = 1;
    private final int mPid;
    private final int mAdPosition;

    private final List<String> mDataset;

    private InReadAdView mAdView;

    /**
     *  Your ad container
     */
    private ViewGroup mAdContainer;

    public SimpleRecyclerViewAdapter(Context context, List<String> dataset, int pid, int adPosition) {
        mPid = pid;
        mAdPosition = adPosition;
        mDataset = dataset;
        mAdContainer = new FrameLayout(context);
        mAdView = new InReadAdView(context);
        mAdView.setPid(mPid);
        mAdView.enableDebug();
        mAdView.load();
    }

    @Override
    public int getItemViewType(int position) {
        return position == mAdPosition ? TYPE_TEADS : TYPE_TEXT;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        switch (viewType) {
            case TYPE_TEADS:
                mAdContainer = parent;
                mAdView.setAdContainerView(mAdContainer);
                return new ViewHolderTeadsAd(mAdView);
            case TYPE_TEXT:
            default:
                View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_row, parent, false);
                return new ViewHolderDemo(v);
        }
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        switch (holder.getItemViewType()) {
            case TYPE_TEADS:
                // loading is already done before hand
                break;
            case TYPE_TEXT:
                ((ViewHolderDemo) holder).textView.setText(mDataset.get(position > mAdPosition && mAdPosition > 0 ?
                                                                                position - 1 :
                                                                                position));
                break;
        }
    }

    @Override
    public int getItemCount() {
        return mDataset.size() + (mAdPosition >= 0 ? 1 : 0);
    }

    public void reloadAd() {
        if (mAdView != null) {
            mAdView.setAdContainerView(mAdContainer);
            mAdView.load();
        }
    }

    private class ViewHolderTeadsAd extends RecyclerView.ViewHolder {

        private ViewHolderTeadsAd(View view) {
            super(view);
        }
    }

    private class ViewHolderDemo extends RecyclerView.ViewHolder {
        private final TextView textView;

        private ViewHolderDemo(View view) {
            super(view);
            textView = view.findViewById(R.id.listViewText);
        }
    }
}

File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/adapter/SimpleRecyclerViewAdapter.kt
package tv.teads.teadssdkdemo.format.adapter

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import tv.teads.sdk.android.AdSettings
import tv.teads.sdk.android.InReadAdView
import tv.teads.teadssdkdemo.R
import tv.teads.teadssdkdemo.data.RecyclerItemType

/**
 * Simple RecyclerView adapter
 */
class SimpleRecyclerViewAdapter(context: Context?, pid: Int)
    : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    // 1. Create the InReadAdView
    private val adView: InReadAdView = InReadAdView(context)

    init {
        // 2. Setup the AdView
        adView.setPid(pid)

        // 3. Create an AdSettings and setup your AdView if needed
        val settings = AdSettings.Builder()
                .enableDebug()
                .build()

        // 4. Load the ad with the created settings
        //    You can still load without settings.
        adView.load(settings)
    }

    override fun getItemViewType(position: Int): Int {
        return when (position) {
            0 -> RecyclerItemType.TYPE_SCROLL_DOWN.value
            1 -> RecyclerItemType.TYPE_ARTICLE_TITLE.value
            2 -> RecyclerItemType.TYPE_ARTICLE_REAL_LINES.value
            4 -> RecyclerItemType.TYPE_TEADS.value
            else -> RecyclerItemType.TYPE_ARTICLE_FAKE_LINES.value
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return when (viewType) {
            RecyclerItemType.TYPE_SCROLL_DOWN.value -> {
                val v = LayoutInflater.from(parent.context).inflate(R.layout.article_header_row, parent, false)
                ViewHolderDemo(v)
            }
            RecyclerItemType.TYPE_ARTICLE_TITLE.value -> {
                val v = LayoutInflater.from(parent.context).inflate(R.layout.article_title_row, parent, false)
                ViewHolderDemo(v)
            }
            RecyclerItemType.TYPE_ARTICLE_REAL_LINES.value -> {
                val v = LayoutInflater.from(parent.context).inflate(R.layout.article_real_lines, parent, false)
                ViewHolderDemo(v)
            }
            RecyclerItemType.TYPE_TEADS.value -> ViewHolderTeadsAd(adView)
            else -> {
                val v = LayoutInflater.from(parent.context).inflate(R.layout.article_fake_lines, parent, false)
                ViewHolderDemo(v)
            }
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {}

    override fun getItemCount(): Int = 6

    private inner class ViewHolderTeadsAd internal constructor(view: View) : RecyclerView.ViewHolder(view)

    private inner class ViewHolderDemo internal constructor(view: View) : RecyclerView.ViewHolder(view) {}
}
